go.mod

```diff
@@ -6,8 +6,11 @@ require (
 	github.com/coreos/dex v2.3.0+incompatible
 	github.com/coreos/go-oidc v2.1.0+incompatible
 	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
+	github.com/devfile/api v0.0.0-20201103130402-29b8738e196e
+	github.com/devfile/library v0.0.0-20201125155652-6eabeab3ee52
 	github.com/gorilla/websocket v1.4.2
 	github.com/graph-gophers/graphql-go v0.0.0-20200309224638-dae41bde9ef9
+	github.com/openshift/api v3.9.0+incompatible
```

Comment:
🤔  I think I tried to remove this but `go mod vendor` added it back I think... I'll try to remove it again.

Reply:
@jhadvig Not sure how to solve this... I think I need more guidance on a solution here.

I swapped it out for the one you quickly suggested on Slack... 
```suggestion
	github.com/openshift/api v0.0.0-20201117184740-859beeffd973
```

And ran what I would think would be the commands:
```
$ go mod vendor
warning: ignoring symlink /.../console/frontend/node_modules/@console/app
...
warning: ignoring symlink /.../console/frontend/node_modules/eslint-plugin-console
go: finding module for package k8s.io/api/auditregistration/v1alpha1
github.com/openshift/console/pkg/helm/chartproxy imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1 imports
	k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.19.4), but does not contain package k8s.io/api/auditregistration/v1alpha1

$ ./build-backend.sh
go: inconsistent vendoring in /.../console:
	golang.org/x/net@v0.0.0-20200707034311-ab3426394381: is explicitly required in go.mod, but vendor/modules.txt indicates golang.org/x/net@v0.0.0-20200421231249-e086a090c8fd
	k8s.io/api@v0.19.2: is explicitly required in go.mod, but vendor/modules.txt indicates k8s.io/api@v0.18.2
	k8s.io/apimachinery@v0.19.2: is explicitly required in go.mod, but vendor/modules.txt indicates k8s.io/apimachinery@v0.18.2

run 'go mod vendor' to sync, or use -mod=mod or -mod=readonly to ignore the vendor directory
```

And it resets the above dep back to what you see there and adds a bunch more lines and fails to compile. Not quite sure how to solve this.

Reply:
It looks like 3.9 API this is coming from here:

https://github.com/devfile/library/blob/master/go.mod#L14

We should try to address this as 3.9 is really, really old.

Reply:
Unfortunately, I don't know enough offhand about Go modules to suggest a fix. @jhadvig might know, or we can ask around.

Reply:
@maysunfaisal Would appear your library is using an old version of the API... perhaps we can update this in your lib? (if I understand the dependency management in go correctly)

Reply:
I can update to openshift api release 4.6 in the library repo.

It seems by default the only publicly available release that go can fetch is 3.9.0. This should probably be updated by the openshift api folks:
```
Maysuns-MacBook-Pro:parser maysun$ go list -m -versions github.com/openshift/api
github.com/openshift/api v3.9.0+incompatible
```

Reply:
I'm a middle man here lol... not sure what the options are. @jhadvig / @spadgett ?

Reply:
I'm updating the library repo to include the 4.6 release with my PR https://github.com/devfile/library/pull/46 so it should be good when that library version is imported here. We may have to wait until then.

Reply:
@maysunfaisal can we bump `openshift/api` it to a higher version then github.com/openshift/api `v0.0.0-20200930075302-db52bc4ef99f` ?

Reply:
@jhadvig that is the commit tagged as OpenShift release 4.6 https://github.com/openshift/api/tree/release-4.6. The release-4.7 branch is even with master and is scheduled for 1Q21.

Reply:
@jhadvig looks bumping to release 4.6 inadvertently updates a lot of other kube packages and it creates a lot of conflicts with the console repo since openshift api uses a lot of the newer versions and console does not. Not sure if this is something we want to do now with this PR. Your advice?

Reply:
These were some of the conflicts and needed updating a lot of console dependencies:
```
go: finding module for package k8s.io/api/auditregistration/v1alpha1
github.com/openshift/console/pkg/helm/actions imports
	k8s.io/client-go/discovery imports
	k8s.io/client-go/kubernetes/scheme imports
	k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.19.4), but does not contain package k8s.io/api/auditregistration/v1alpha1
```

```
vendor/k8s.io/cli-runtime/pkg/resource/dry_run_verifier.go:23:2: case-insensitive import collision: "github.com/googleapis/gnostic/OpenAPIv2" and "github.com/googleapis/gnostic/openapiv2"
```

```
# helm.sh/helm/v3/pkg/kube
vendor/helm.sh/helm/v3/pkg/kube/client.go:180:26: too many arguments in call to helper.Get
vendor/helm.sh/helm/v3/pkg/kube/client.go:180:58: info.Export undefined (type *"k8s.io/cli-runtime/pkg/resource".Info has no field or method Export)
vendor/helm.sh/helm/v3/pkg/kube/client.go:380:31: too many arguments in call to helper.Get
vendor/helm.sh/helm/v3/pkg/kube/client.go:380:69: target.Export undefined (type *"k8s.io/cli-runtime/pkg/resource".Info has no field or method Export)
vendor/helm.sh/helm/v3/pkg/kube/client.go:485:11: undefined: "k8s.io/client-go/tools/watch".ListWatchUntil
```

Reply:
I was able to resolve these conflicts but its results in updating all these deps and vendor dir:

![Screen Shot 2020-12-03 at 4 00 55 PM](https://user-images.githubusercontent.com/31771087/101090461-b09d8300-3584-11eb-915e-18268ae9abc9.png)